File: src/constants/strings.py
"""Strings constant"""

from __future__ import annotations

from enum import StrEnum


class Strings:
    """Strings definitions"""

    class Misc(StrEnum):
        PRESS_START = "Press any key to continue"

    class Menu(StrEnum):
        NEW_GAME = "New Game"
        RESUME = "Resume"
        QUIT_TO_DESKTOP = "Quit to Desktop"
        SAVE_AND_QUIT = "Save and Quit"
        QUIT_NO_SAVE = "Quit without Saving"

    class Titles(StrEnum):
        BESTIARY = "bestiary"
        LEVEL_UP = "level up"
        INVENTORY = "inventory"
        PAUSE = "pause"
        GAME = "Satan's Lil Helper"
        MAIN_MENU = GAME
        STATUS = "status"
        CHARACTER_SHEET = "character info"

    class Status(StrEnum):
        LOCATION = "location"
        HEALTH = "health"
        MANA = "mana"
        ABILITIES = "abilities"
        STRENGTH = "strength"
        MAGIC = "magic"
        EVASION = "evasion"
        CRIT = "crit"
        EQUIPMENT = "equipment"
        LIGHT = "light radius"
        VISION = "vision radius"
        SKILLS = "skills"

    class GearSlots(StrEnum):
        HEAD = "helm"
        BODY = "armor"
        HANDS = "gauntlets"
        FEET = "boots"
        WEAPON = "weapon"
        

File: src/Constants/Strings.cpp
#include "Constants.hpp"
#define _(STRING) translate(STRING)
#define N_(STRING) STRING
#define C_(CONTEXT, STRING) translate(CONTEXT, STRING)

namespace SatansLilHelper::Constants::Strings {

    extern const basic_message<char> Title      = _("Satan's Lil Helper");
    extern const basic_message<char> PressStart = _("Press START");

    namespace Menu {
        extern const basic_message<char> NewGame
            = _("new game");
        extern const basic_message<char> Bestiary
            = _("bestiary");
        extern const basic_message<char> Settings
            = _("settings");
    }

}
